src/sql/src/plan/lowering.rs

```diff
@@ -1148,6 +1285,39 @@ where
             });
         }
     });
+    // Collect all the outer columns referenced by any CTE referenced by
+    // the inner relation.
+    inner.visit(&mut |e| match e {
+        HirRelationExpr::Get {
+            id: expr::Id::Local(id),
+            ..
+        } => {
+            if let Some(cte_desc) = cte_map.get(id) {
```

Comment:
In what case would `cte_map` not contain an entry for `id`?

Reply:
For example right before lowering a correlated scalar subquery with a CTE. We may find references within the subquery to the CTE defined within it, but we haven't visited that CTE, because that happens while lowering the subquery.